sauce/features/accounts/customFlagNames/index.js

```diff
@@ -0,0 +1,148 @@
+let orangeFlagLabel;
+let yellowFlagLabel;
+let greenFlagLabel;
+let purpleFlagLabel;
+
+export default class CustomFlagNames extends Feature {
+  constructor() {
+    super();
+    if (!localStorage.getItem('flags')) {
+      this.storeDefaultFlags();
+    }
+    if (typeof flags === 'undefined') {
+      flags = JSON.parse(localStorage.getItem('flags'));
+      this.updateFlagLabels();
+    }
+  }
+
+  shouldInvoke() {
+    return toolkitHelper.getCurrentRouteName().indexOf('account') !== -1;
+  }
+
+  invoke() {
+    $('.ynab-grid-cell-flag .ynab-flag-red').parent().attr('title', redFlagLabel);
+    $('.ynab-grid-cell-flag .ynab-flag-blue').parent().attr('title', blueFlagLabel);
+    $('.ynab-grid-cell-flag .ynab-flag-orange').parent().attr('title', orangeFlagLabel);
+    $('.ynab-grid-cell-flag .ynab-flag-yellow').parent().attr('title', yellowFlagLabel);
+    $('.ynab-grid-cell-flag .ynab-flag-green').parent().attr('title', greenFlagLabel);
+    $('.ynab-grid-cell-flag .ynab-flag-purple').parent().attr('title', purpleFlagLabel);
+  }
+
+  observe(changedNodes) {
+    if (!this.shouldInvoke()) return;
+
+    if (changedNodes.has('ynab-flag-red') || changedNodes.has('ynab-flag-blue')
+      || changedNodes.has('ynab-flag-orange') || changedNodes.has('ynab-flag-yellow')
+      || changedNodes.has('ynab-flag-green') || changedNodes.has('ynab-flag-purple')) {
+      this.invoke();
+    }
+
+    if (changedNodes.has('ynab-u modal-popup modal-account-flags ember-view modal-overlay active')) {
+      $('.ynab-flag-red .label, .ynab-flag-red .label-bg').html(redFlagLabel);
+      $('.ynab-flag-blue .label, .ynab-flag-blue .label-bg').html(blueFlagLabel);
+      $('.ynab-flag-orange .label, .ynab-flag-orange .label-bg').html(orangeFlagLabel);
+      $('.ynab-flag-yellow .label, .ynab-flag-yellow .label-bg').html(yellowFlagLabel);
+      $('.ynab-flag-green .label, .ynab-flag-green .label-bg').html(greenFlagLabel);
+      $('.ynab-flag-purple .label, .ynab-flag-purple .label-bg').html(purpleFlagLabel);
+
+      $('.modal-account-flags .modal').css({ height: '22em' })
+        .append('<div id="account-flags-actions" style="padding: 0 .3em"><button id="flags-edit" class="button button-primary">Edit <i class="flaticon stroke compose-3"><!----></i></button></div>');
+
+      this.addEventListeners();
+    }
+  }
+
+  onRouteChanged() {
+    if (!this.shouldInvoke()) return;
+
+    this.invoke();
+  }
+
+  addEventListeners() {
+    let $this = this;
+    $('#flags-edit').click(function () {
+      $('.modal-account-flags .modal-list').empty();
+
+      for (let key in flags) {
+        let flag = flags[key];
+        $('.modal-account-flags .modal-list').append('<li><input type="text" id="' + key + '" class="flag-input" style="color: #fff; fill: ' + flag.color + '; background-color: ' + flag.color + '; height: 30px; padding:0 .7em; margin-bottom: .3em; border: none;" value="' + flag.label + '" placeholder="' + flag.label + '" /></li>');
```

Comment:
Thank you both for the info. I learned something new regarding jQueries append with this PR.